launchdarkly/resource_feature_flag.go

```diff
@@ -0,0 +1,250 @@
+	d.Set("tags", tags)
+	d.Set("custom_properties", customProperties)
+
+	return nil
+}
+
+func resourceFeatureFlagRead(d *schema.ResourceData, m interface{}) error {
+	project := d.Get("project_key").(string)
+	key := d.Get("key").(string)
+
+	client := m.(Client)
+
+	payload := make(map[string]interface{})
+
+	err := client.Get("/flags/"+project+"/"+key, map[int]bool{200: true}, &payload)
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	transformedTags := payload["tags"].([]interface{})
+	transformedCustomProperties := transformCustomPropertiesFromLaunchDarklyFormat(payload["customProperties"])
+
+	d.Set("name", payload["name"])
+	d.Set("key", payload["key"])
+	d.Set("description", payload["description"])
+	d.Set("temporary", payload["temporary"])
+	d.Set("include_in_snippet", payload["includeInSnippet"])
+	d.Set("tags", transformedTags)
+	d.Set("custom_properties", transformedCustomProperties)
+
+	return nil
+}
+
+func resourceFeatureFlagUpdate(d *schema.ResourceData, m interface{}) error {
+	client := m.(Client)
+
+	project := d.Get("project_key").(string)
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	temporary := d.Get("temporary").(bool)
+	includeInSnippet := d.Get("include_in_snippet").(bool)
+	tags := d.Get("tags").([]interface{})
+	customProperties := d.Get("custom_properties").([]interface{})
+
+	transformedCustomProperties, err := transformCustomPropertiesFromTerraformFormat(customProperties)
+	if err != nil {
+		return err
+	}
+
+	payload := []map[string]interface{}{{
+		"op":    "replace",
+		"path":  "/name",
+		"value": name,
+	}, {
+		"op":    "replace",
+		"path":  "/description",
+		"value": description,
+	}, {
+		"op":    "replace",
+		"path":  "/temporary",
+		"value": temporary,
+	}, {
+		"op":    "replace",
+		"path":  "/includeInSnippet",
+		"value": includeInSnippet,
+	}, {
+		"op":    "replace",
+		"path":  "/tags",
+		"value": transformTagsFromTerraformFormat(tags),
+	}, {
+		"op":    "replace",
+		"path":  "/customProperties",
+		"value": transformedCustomProperties,
+	}}
+
+	err = client.Patch("/flags/"+project+"/"+d.Id(), payload, map[int]bool{200: true}, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceFeatureFlagDelete(d *schema.ResourceData, m interface{}) error {
+	client := m.(Client)
+
+	project := d.Get("project_key").(string)
+
+	err := client.Delete("/flags/"+project+"/"+d.Id(), map[int]bool{204: true, 404: true})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func transformTagsFromTerraformFormat(tags []interface{}) []string {
+	transformed := make([]string, len(tags))
+
+	for index, value := range tags {
+		transformed[index] = value.(string)
+	}
+
+	return transformed
+}
+
+func transformCustomPropertiesFromTerraformFormat(properties []interface{}) (interface{}, error) {
+	transformed := make(map[string]interface{})
+
+	for _, value := range properties {
+		key := value.(map[string]interface{})["key"].(string)
```

Comment:
You should probably only assert type once. It has a small performance overhead and it is more error prone to do it on very line.

Reply:
Indeed, I was lazy. Fixed.